_posts/2020-09-30-hows-the-performance-of-ruby-3.0.0-preview1.md

```diff
@@ -0,0 +1,80 @@
+---
+layout: post
+title: "How's the Performance of Ruby 3.0.0-preview1?"
+date: 2020-09-30 7:00:00
+categories: ["rails", "performance", "ruby"]
+author: ["noahgibbs"]
+---
+
+The [new Ruby 3.0 preview](https://www.ruby-lang.org/en/news/2020/09/25/ruby-3-0-0-preview1-released/) is out! Woo-hoo!
+
+If you've heard of me, you know [performance is kinda my thing](https://engineering.appfolio.com/?author=5751bf4722482e6c3dbfc424), especially [Rails performance on large apps](https://github.com/noahgibbs/rails_ruby_bench). I do other stuff too, but I got paid to do that for years (thanks, AppFolio!), so I've written a lot about it.
+
+How does the new preview's performance stack up on Rails? And how reliable are these numbers?
+
+<!--more-->
+
+## Pitfalls
+
+First off, not every gem is ready for Ruby 3. For instance, the latest version of ruby_dep (1.5.0) has a "\~>2.2" dependency on Ruby. I'm not trying to pick on it! [Tilde-dependencies are usually a really good idea](https://guides.rubygems.org/patterns/#pessimistic-version-constraint)! And now a bunch of them are going to need to change.
+
+I'm also using ancient code for this benchmark, frankly. I consider myself a pragmatist on this point, and I tried what a pragmatist would do. &lt;whispering&gt;I commented out the check for Ruby version in the local copy of Bundler 1.1.17&lt;/whispering&gt;. Um, I mean we're all respectable Rubyists here who never cut corners to get speed ratings of prerelease software.
+
+It didn't work, though. It turns out there are ***several*** changes in Ruby 3.0 that are going to require gems to upgrade a bit. But if the main problem is just the version number&hellip;
+
+I put together [a Ruby branch which started out exactly identical to the Git SHA for Ruby 3.0.0-preview1, plus a commit that rolled the reported version back to 2.8](https://github.com/noahgibbs/ruby/tree/fake_2_8). That's not as good an idea as updating all the gems to support 3.0, but it requires a lot fewer updates to ancient software and/or that I can't do for myself. Though I still have to revert a few deprecations so that old code warns instead of crashing...
```

Comment:
```suggestion
I put together [a Ruby branch which started out exactly identical to the Git SHA for Ruby 3.0.0-preview1, plus a commit that rolled the reported version back to 2.8](https://github.com/noahgibbs/ruby/tree/fake_2_8). That's not as good an idea as updating all the gems to support 3.0, but it requires a lot fewer updates to ancient software  that I can't do for myself. Though I still have to revert a few deprecations so that old code warns instead of crashing...
```

Maybe I am missing something, but I wasn't sure what the **and/or** is referring to here.